File: ros2_ws/src/example_python/example_python/pub_node.py
import rclpy
from rclpy.node import Node 
from interfaces.msg import Example

class PubNode(Node):

    def __init__(self):
        super().__init__('pub_node')
        self.publisher = self.create_publisher(Example, "ExampleTopic", 10)
        self.timer = self.create_timer(3.0, self.timer_callback)
        self.count = 0

    def timer_callback(self):
        self.count += 1
        example = Example()
        example.message = self.count
        self.publisher.publish(example)

def main():
    rclpy.init()
    node = PubNode()
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        pass
    finally:
        node.destroy_node()
        rclpy.shutdown()

if __name__ == '__main__':
    main()

File: ros2_ws/src/example_cpp/src/pub_node.cpp
#include "interfaces/msg/example.hpp"
#include "rclcpp/rclcpp.hpp"

using namespace std::chrono_literals;

class PubNode : public rclcpp::Node
{
public:
    PubNode() : Node("pub_node")
    {
        publisher_ = this->create_publisher<interfaces::msg::Example>("ExampleTopic", 10);
        timer_ = this->create_wall_timer(3s, std::bind(&PubNode::timer_callback, this));
        count_ = 0;
    }

private:
    void timer_callback()
    {
        count_++;
        auto example = interfaces::msg::Example();
        example.message = count_;
        RCLCPP_INFO(this->get_logger(), "Publishing: %i", example.message);
        publisher_->publish(example);
    }

    rclcpp::TimerBase::SharedPtr timer_;
    rclcpp::Publisher<interfaces::msg::Example>::SharedPtr publisher_;
    int count_;
};

int main(int argc, char *argv[])
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<PubNode>());
    rclcpp::shutdown();
    return 0;
}